sotopia/generation_utils/generate.py

```diff
@@ -447,7 +447,7 @@ async def agenerate(
     output_parser: BaseOutputParser[OutputType],
     temperature: float = 0.7,
 ) -> OutputType:
-    input_variables = re.findall(r"{(.*?)}", template)
+    input_variables = re.findall(r"(?<!{){([^{}]+)}(?!})", template)
```

Comment:
Can you add some examples to the doc to illustrate why this is desirable? It seems that it ignores all nested curly parentheses. What is the reason behind this?

Reply:
will write in the doc string

Reply:
updated (just the in-line explanation for now)
Will update the doc more in detail in the `dev` branch!